cdap-api/src/main/java/co/cask/cdap/api/plugin/EndpointPluginContext.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.plugin;
+
+import co.cask.cdap.api.annotation.Beta;
+
+/**
+ * Provides access to instantiate plugin for plugin endpoint methods.
+ */
+@Beta
+public interface EndpointPluginContext {
```

Comment:
here we want to find the plugin and then load the plugin, combination of what plugin configurer and plugin context does currently, do you mean we aren't extending PluginContext ?